content/starter-configs/tutorials/wordpress.md

```diff
@@ -48,62 +50,106 @@ define('DB_PASSWORD', 'tugboat');
     #   3. Routes requests to the preview URL to this service
     default: true
 
-    # Wait until the mysql service is done building
+    # Wait until the mysql service is done building so we have a database.
     depends: mysql
 
-    # A set of commands to run while building this service
+    # A set of commands to run while building this service.
     commands:
-      # Commands that set up the basic preview infrastructure
+      # Phase 1 (init): Configure the server infrastructure.
       init:
-        # Install prerequisite packages
+        # Install prerequisite packages.
         - apt-get update
-        - apt-get install -y rsync libzip-dev
+        - apt-get install -y rsync
 
-        # Enable extra apache modules.
-        - a2enmod rewrite headers
+        # Turn on URL rewriting.
+        - a2enmod rewrite
 
-        # Install the PHP extensions
-        - docker-php-ext-install mysqli exif zip
-        # Install imagick
-        - apt-get install -y libmagickwand-dev
-        - pecl install imagick-beta -y
-        - docker-php-ext-enable imagick
+        # Install PHP mysqli extension.
+        - docker-php-ext-install mysqli
 
-        # Install wp-cli
+        # Install the Wordpress CLI tool.
         - curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
         - chmod +x wp-cli.phar
         - mv wp-cli.phar /usr/local/bin/wp
 
-        # Use the tugboat-specific wp-config.local.php
-        - cp "${TUGBOAT_ROOT}/.tugboat/wp-config.local.php" "${DOCROOT}/"
+        # If using composer to install Wordpress, uncomment this line.
+        # - composer install --optimize-autoloader
 
-        # Link the document root to the expected path. This example links /web
-        # to the docroot
-        - ln -snf "${TUGBOAT_ROOT}/web" "${DOCROOT}"
+        # Link Wordpress core as the docroot.
+        - ln -snf "${TUGBOAT_ROOT}/docroot" "${DOCROOT}"
 
-      # Commands that import files, databases, or other assets. When an
-      # existing preview is refreshed, the build workflow starts here,
-      # skipping the init step, because the results of that step will
-      # already be present.
+        # Copy and link the Tugboat config file into the Wordpress core directory.
+        - rm -rf ${TUGBOAT_ROOT}/wp-config.php; rm -rf ${DOCROOT}/wp-config.php; cp
+          ${TUGBOAT_ROOT}/.tugboat/wp-config.tugboat.php ${TUGBOAT_ROOT}/wp-config.php
+        - ln -snf ${TUGBOAT_ROOT}/wp-config.php ${DOCROOT}/wp-config.php
+
+        # Install wp-cli.
+        - curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+        - chmod +x wp-cli.phar
+        - mv wp-cli.phar /usr/local/bin/wp
+
+        # Update permalinks to remove the index.php.  These can't run until the database is imported.  How do we do that?
+        - wp --allow-root --path="${DOCROOT}" option set permalink_structure /%postname%/
+        - wp --allow-root --path="${DOCROOT}" rewrite flush --hard
```

Comment:
> These can't run until the database is imported.  How do we do that?

I would move these lines to the `update` phase to ensure the database is imported first.

Reply:
We have the `php` service set as a dependency of `mysql`, so the database does get imported first.  Then we can leave these other commands in the INIT to reduce refresh times and new builds from base previews.